tests/Metadata/Driver/DocBlockDriverTest.php

```diff
@@ -131,6 +131,37 @@ public function testInferDocBlockMapFromGenericLikeClass()
             ['name' => 'array', 'params' => [['name' => 'int', 'params' => []], ['name' => Product::class, 'params' => []]]],
             $m->propertyMetadata['productIds']->type,
         );
+        self::assertEquals(
+            [
+                'name' => 'array',
+                'params' => [
+                    ['name' => Product::class, 'params' => []],
+                    ['name' => Vehicle::class, 'params' => []],
+                ],
+            ],
+            $m->propertyMetadata['productOrVehicleIds']->type,
+        );
+        self::assertEquals(
+            [
+                'name' => 'array',
+                'params' => [
+                    ['name' => 'int', 'params' => []],
+                    ['name' => Product::class, 'params' => []],
+                    ['name' => Vehicle::class, 'params' => []],
```

Comment:
@simPod It seems like `array<int, string>` and `array<int|string>` will create exactly the same metadata 😓 I would consider using `Union` pseudo type same as for the unions, so it will be handled by the Union Handler :D